CenterForOpenScience/osf.io
# [ENG-373] Create periodic task that checks and sets stuck DOIs

## Purpose

We've had several DOI confirmation emails that may have been lost during downtime. This PR will check daily to see if any DOI confirmations have been missed.   

## Changes

- add periodic script and tests

## QA Notes

Preprints that are missing DOI confirmation (there's no link to the preprint's DOI) - should appear at 11 PM.   

In summary, if we have published preprints, without the `preprint_doi_created` set, we filter crossref's API for these DOI's.  If they exist, we set the DOI in our db.  Otherwise, we email the help desk.

## Documentation

Not user facing, no docs.

## Side Effects

None that I know of.

## Ticket

https://openscience.atlassian.net/browse/ENG-373
